src/App.jsx

```diff
@@ -1,35 +1,48 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useState, useEffect } from "react";
+import { Routes, Route, useLocation } from "react-router";
+import axios from "axios";
+
+import Header from "./components/Header";
+import Home from "./components/Home";
+import Articles from "./components/Articles";
+import Topics from "./components/Topics";
+import Contacts from "./components/Contacts";
+
+import "./App.css";
 
 function App() {
-  const [count, setCount] = useState(0)
+  // const { pathname } = useLocation();
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    axios
+      .get("https://nc-news-api-qfui.onrender.com/api/articles")
+      .then(({ data }) => {
+        setArticles(data.articles);
+        setIsLoading(false);
+      });
+  }, []);
```

Comment:
This state and useEffect should really be in your Articles component if possible. Having it in your app means that every time there's a change, you're also going to be re-rendering everything in there, such as static components like your Header.